### PR TITLE
Implemented possible fix for #2370

### PR DESCRIPTION
**Pull request by oshtaier**
Removing unnecessary code and dependencies. Added one more verification to reduce amount of possible root causes for initial failure

### CODE REVIEW

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/api/test_hostgroup.py -m test_bz_1107708
.
----------------------------------------------------------------------
Ran 1 test in 93.395s

OK
```

**Comment by oshtaier** [comment]
Close initial issue only if test has few successful runs for Compose builds

File: tests/foreman/api/test_hostgroup.py [hunk 1]
```diff
@@ -36,26 +33,24 @@ def test_bz_1107708(self):
...
             content_type='puppet',
             product=product,
-            url=FAKE_0_PUPPET_REPO,
         ).create()
-        puppet_repo.sync()
+        # Working with 'ntp' module as we know for sure that it contains at
+        # least few puppet classes
         with open(get_data_file(PUPPET_MODULE_NTP_PUPPETLABS), 'rb') as handle:
             puppet_repo.upload_content(handle)
+        puppet_repo.sync()
```
Comment (omaciel) [inline tests/foreman/api/test_hostgroup.py]: You don't need to sync it since you've already uploaded it

**Comment by omaciel** [comment]
Your fix is sort of a work around since we've gone from a total of 5 puppet module to 2 and we should then be able to add it to the content view. Perhaps we should then create a new test to check that the previous scenario works? In other words, a test where you can add a specific puppet module to a content when there are several available. This would go outside the smoke suite.

**Comment by oshtaier** [comment]
"from a total of 5 puppet module to 2" to 1 actually

If that scenario work, definitely we can look to expand it, but I really prefer to work with a things that I can, at least, debug. As I got 347 successful runs in a row, it will be quite hard for me to find out real root cause for the problem which can be reproducible only on Jenkins.

Let's narrow the place of possible failure first

**Comment by omaciel** [comment]
I meant to type "from 5 to 1" but my fingers are too big for my phone :)

ACK

**Comment by Ichimonji10** [comment]
Everything is syntactically great. ACK

> As I got 347 successful runs in a row, it will be quite hard for me to find out real root cause for the problem which can be reproducible only on Jenkins.

One difference I've noticed between running tests manually and running them on Jenkins is that, when the latter is happening, load-related failures occur. The underlying host we use for our tests runs several (up to four?) Satellite instances at once, and each of those Sat instances may be responding to an hours-long series of tests.

**Comment by elyezer** [comment]
I missed this, but this issue is caused by https://bugzilla.redhat.com/show_bug.cgi?id=1222118.

Not being able to reproduce locally is because you don't have other already synced puppet modules on your test machine. On jenkins machines happens because it run many tests that syncs puppet modules and as the filter is not working it returns more than it should.

**Comment by Ichimonji10** [comment]
Thanks for the explanation, @elyezer.

**Pull request merged**